docs/excel/excel-add-ins-conditional-formatting.md

```diff
@@ -0,0 +1,305 @@
+
+![A range with databars behind the values in cells.](../images/excel-conditional-format-databar.png)
+
+```typescript
+const sheet = context.workbook.worksheets.getItem("Sample");
+const range = sheet.getRange("B8:E13");
+const conditionalFormat = range.conditionalFormats.add(
+     Excel.ConditionalFormatType.dataBar
+);
+
+// give left-to-right data bars to all the cells, with the min and max adjusted to the range
+conditionalFormat.dataBar.barDirection = Excel.ConditionalDataBarDirection.leftToRight;
+await context.sync();
+```
+
+### [Icon set](https://docs.microsoft.com/javascript/api/excel/excel.iconsetconditionalformat)
+
+Icon set formatting uses Excel [Icons]( https://docs.microsoft.com/javascript/api/excel/excel.icon) to highlight cells. The `criteria` property is an array of [ConditionalIconCriterion](https://docs.microsoft.com/javascript/api/excel/excel.ConditionalIconCriterion), which define the symbol to inserted and the conditional under which it is inserted. This array is automatically prepopulated with criterion elements with default properties. Individual properties cannot be overwritten. Instead, the whole criteria object must be replaced. The following example shows a three-triangle icon set applied across the range.
+
+![A range with green upward triangles for values above 1000, yellow lines for valuese between 700 and 1000, and red downward triangles for lower values.](../images/excel-conditional-format-iconset.png)
+
+```typescript
+const sheet = context.workbook.worksheets.getItem("Sample");
+const range = sheet.getRange("B8:E13");
+const conditionalFormat = range.conditionalFormats.add(
+     Excel.ConditionalFormatType.iconSet
+);
+
+const iconSetCF = conditionalFormat.iconSet;
+iconSetCF.style = Excel.IconSet.threeTriangles;
+
+/*
+   With a "three*” icon set style, such as "threeTriangles", the third
+    element in the criteria array (criteria[2]) defines the "top" icon;
+    e.g., a green triangle. The second (criteria[1]) defines the "middle"
+    icon, The first (criteria[0]) defines the "low" icon, but it can often 
+    be left empty as this method does below, because every cell that
+   does not match the other two criteria always gets the low icon.
+*/
+iconSetCF.criteria = [
+    {} as any,
+      {
+        type: Excel.ConditionalFormatIconRuleType.number,
+        operator: Excel.ConditionalIconCriterionOperator.greaterThanOrEqual,
+        formula: "=700"
+      },
+      {
+        type: Excel.ConditionalFormatIconRuleType.number,
+        operator: Excel.ConditionalIconCriterionOperator.greaterThanOrEqual,
+        formula: "=1000"
+      }
+];
+
+await context.sync();
+```
+
+### [Preset criteria](https://docs.microsoft.com/javascript/api/excel/excel.presetcriteriaconditionalformat)
```

Comment:
To be consistent with the opening sentence of other like paragraphs, consider starting this one with "Preset criteria conditional formatting..."